src/include/network/lan/hardware.rb

```diff
@@ -51,19 +51,26 @@ def initialize_network_lan_hardware(include_target)
       Yast.include include_target, "network/lan/cards.rb"
 
       @hardware = nil
+    end
 
-      @widget_descr_hardware = {
-        "HWDIALOG" => {
-          "widget"            => :custom,
-          "custom_widget"     => ReplacePoint(Id(:hw_content), Empty()),
-          "init"              => fun_ref(method(:initHwDialog), "void (string)"),
-          "handle"            => fun_ref(method(:handleHW), "symbol (string, map)"),
-          "store"             => fun_ref(method(:storeHW), "void (string, map)"),
-          "validate_type"     => :function,
-          "validate_function" => fun_ref(method(:validate_hw), "boolean (string, map)"),
-          "help"              => initHelp
-        }
+    def widget_descr_hardware
+      widget_descr = {
+        "widget"        => :custom,
+        "custom_widget" => ReplacePoint(Id(:hw_content), Empty()),
+        "init"          => fun_ref(method(:initHwDialog), "void (string)"),
+        "handle"        => fun_ref(method(:handleHW), "symbol (string, map)"),
+        "store"         => fun_ref(method(:storeHW), "void (string, map)"),
+        "help"          => initHelp
       }
+
+      # validation function currently checks user's input in :ifcfg_name widget
+      # However this widget is present only when adding new device
+      if isNewDevice
```

Comment:
short summary:
- ```validate_hw``` is not needed in case of editing device
- always return ```true``` is thing I wanted to avoid. Validation function which always returns true is nonsense from my POV
- current place was chosen to provide widget definition according its purpose in time to avoid various widgets removal / adding during init phase.